boot/bootstate20.go

```diff
@@ -720,3 +720,44 @@ func trustedAssetsBootState(dev Device) *bootState20BootAssets {
 		dev: dev,
 	}
 }
+
+// bootState20CommandLine implements the successfulBootState interface for
+// kernel command line
+type bootState20CommandLine struct {
+	dev Device
+}
+
+func (ba20 *bootState20CommandLine) markSuccessful(update bootStateUpdate) (bootStateUpdate, error) {
+	u20, err := toBootStateUpdate20(update)
+	if err != nil {
+		return nil, err
+	}
+	if len(u20.modeenv.CurrentTrustedBootAssets) == 0 && len(u20.modeenv.CurrentTrustedRecoveryBootAssets) == 0 {
+		// XXX: does this change when we expose the ability to add
+		// things to the command line?
+
+		// not using trusted boot assets, bootloader config is not
+		// managed and command line can be manipulated externally
+		return update, nil
+	}
+
+	var newM *Modeenv
+	if len(u20.modeenv.CurrentKernelCommandLines) == 0 {
+		// no command lines tracked in modeenv yet, this can happen when
+		// having booted with a newer snapd
+		newM, err = observeSuccessfulCommandLineCompatBoot(ba20.dev.Model(), u20.writeModeenv)
```

Comment:
I see we do the same with observeSuccessfulBootAssets,  I didn't wonder before but should these take as input writeModeenv or modeenv, it's a bit unclear to me right now?  if they really take writeModeenv do they need to make a copy?